### ISSUE TITLE
Location of database

### ISSUE THREAD

**Issue by kovasap** [ActivityWatch/aw-android#62]
Is the database saved as a sqlite file somewhere on the device I could get access to?  I'm looking for a way to automatically backup my usage data from my phone without having to manually use the export feature.  I have another app that lets me sync specific directories on my phone to the cloud; I'd like to feed it the AW data directory.  

Does this make sense or is there a better way to achieve what I'm looking for?

**Comment by ErikBjare** [comment]
The Android app stores the SQLite file in its [assigned data directory](https://android.stackexchange.com/questions/47924/where-android-apps-store-data), however this directory is likely protected from being accesses by other apps on the file system (for security reasons). However, if you've rooted your phone it should be easy to access, and you might also be able to access it with adb.

**Comment by kovasap** [comment]
Ah ok I see.  I see that AW also has a directory in `/storage/emulated/0/Android/data/net.activitywatch.android`, but it is empty.  Could you point me to the place in the code where the database file location is set (or is auto set by default)?  I think for me it would be easiest to change this in a aw-android fork (or contribute it back if you would like).  I can't root my current phone as I use it for work as well : (.

**Comment by kovasap** [comment]
Nvm, see https://github.com/ActivityWatch/aw-android/pull/63

**Comment by kovasap** [comment]
If it makes more sense, I could also try to accomplish this by adding a auto-export feature to the server, which would work for all AW platforms.  What do you think?

**Comment by kanishkarj** [comment]
Hi, it might also work if the app could auto-expory to Dropbox or some cloud storage once a day I guess. I don't think we would need realtime sync, we just need the ability to consolidate data from all devices in one location right?

**Comment by kovasap** [comment]
I was hoping to fix this problem for myself by using adb to [make just the acitivitywatch data directory readable by all](https://stackoverflow.com/questions/51042115/how-to-change-permission-of-internal-folder-file-in-android), but this also requires root : (.  

I did discover that running `adb backup net.activitywatch.android` when the device is connected to the PC with developer mode enabled produces a `backup.ab` file.  You can then extract this file with `( printf "\x1f\x8b\x08\x00\x00\x00\x00\x00" ; tail -c +25 backup.ab ) |  tar xfvz -` to get at all the data activitywatch stores, including the `sqlite.db` file.  Perhaps there is a way to run this command on the phone itself via [LADB](https://forum.xda-developers.com/t/say-hello-to-ladb-a-local-adb-shell-without-needing-root-or-a-computer.4204855/), but this seems to require Android 11 for the WiFi debugging, which I don't have.  If this is possible, you could periodically run the backup command to get the data exported automatically to a readable dir on your phone, then sync that dir to wherever.  

Anyway, the best solution I think would be to either add an option to have activitywatch automatically export, or have the [syncing feature](https://github.com/ActivityWatch/activitywatch/issues/35) implemented so that this data is pushed to my pc and I can just get it from there.  Or you could just root your phone : ).

**Comment by khaalidrafi** [comment]
Is it possible now to access the sqlite db without root?

**Comment by ErikBjare** [comment]
@kanishkarj Unlikely, this is an Android security "feature" that isn't going away.

**Comment by khaalidrafi** [comment]
> Unlikely, this is an Android security "feature" that isn't going away.

Is it possible to place the file elsewhere from the app, where we have permissions? The db is important for making plugins. It helps make well-customized queries for data. And, our Android data is important since we spend a lot of time here. The only good activity tracker I found is StayFree, which isn't open source. This app needs to improve.

**Comment by ErikBjare** [comment]
@khalidrafi6 You're right of course, it was proposed in https://github.com/ActivityWatch/aw-android/pull/63

However I'm concerned that such a change may not pass Play Store review (not that it should stop us, just a worry I have, I'm probably just overly cautious from the experience of past failed reviews).

There's a lot of improvements to be done on the Android app, for sure. @0xbrayo has been working on improvements like adding sync-support to the Android app, which may be the preferred way to do this.
